torch/_dynamo/side_effects.py

```diff
@@ -22,42 +22,75 @@
 from .utils import is_frozen_dataclass, nn_module_new, object_new
 from .variables.base import (
     is_side_effect_safe,
-    MutableLocalBase,
-    MutableLocalSource,
+    MutationType,
+    SourceType,
     VariableTracker,
 )
 from .variables.user_defined import FrozenDataClassVariable
 
 
-class MutableSideEffects(MutableLocalBase):
+class ValueMutationExisting(MutationType):
```

Comment:
Move these to base.py? Thinking if all these are better to have in one file.

Reply:
It's in the next patch, wanted to make review easier:) #139340